[1] feat(lib-agnostic): migrate latest SDK changes

shoom3301 opened the pull request:
Migrated all packages besides **BRIDGING**, because this package contains the largest count of changes.
The Bridging package is migrated here: https://github.com/cowprotocol/cow-sdk/pull/426

Some tests are broken, but they are fixed in the Bridging changes PR, which is open on top of the current one.

## How the migration is done
1. I went through every change in this draft: https://github.com/cowprotocol/cow-sdk/pull/425, which is a diff between `main` <> `SDK-AGNOSTIC-LIB` 
2. Copy-pasted every change to the new version
3. Adjusted code if needed
4. Added new Adapter methods when needed

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- New Features
  - Added AppData v1.5.0/v1.6.0 with bridging and flashloan metadata support.
  - Enabled BNB and Lens networks across configuration and Order Book API.
  - Introduced SigningStepManager hooks to orchestrate pre/post signing steps.
  - CowShed SDK now versioned with required deadline and optional gasLimit.

- Improvements
  - Dynamic, chain-aware default slippage and deadline handling; refined EthFlow address resolution and gas estimation.
  - Bridging updates: increased default gas estimate and new hook provider prefix.
  - Providers now expose code/receipt retrieval; unified bigint blocks/logs and added log parsing utilities.
  - Updated chain labels and explorers for clearer network identification.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

alfetopito commented on packages/trading/src/postCoWProtocolTrade.ts:
```diff
@@ -69,45 +68,3 @@ export async function postCoWProtocolTrade(
   return { orderId, signature, signingScheme, orderToSign }
 }
-
-async function createAppDataWithUTM(originalAppData: TradingAppDataInfo): Promise<TradingAppDataInfo> {
```
Why remove `createAppDataWithUTM`?

jeffersonBastos replied:
Just for record-keeping and to keep it tracked, this function came from the issue: https://github.com/cowprotocol/cow-sdk/issues/388

shoom3301 replied:
https://github.com/cowprotocol/cow-sdk/issues/388#issuecomment-3210064229

alfetopito commented on packages/app-data/src/schemas/v1.6.0.json:
```diff
@@ -0,0 +1,84 @@
+{
+  "$id": "https://cowswap.exchange/schemas/app-data/v1.5.0.json",
```
I know it's not from here, but the schema has a bug: should be 1.6.0

shoom3301 replied:
Fixed

alfetopito commented on packages/app-data/src/schemas/v1.6.0.json:
```diff
@@ -0,0 +1,84 @@
+    "version": {
+      "$ref": "definitions.json#/definitions/version",
+      "readOnly": true,
+      "default": "1.5.0"
```
Same here

alfetopito commented on packages/config/src/constants/paths.ts:
```diff
@@ -3,7 +3,3 @@ export const RAW_FILES_PATH = 'https://raw.githubusercontent.com/cowprotocol/cow
 export const RAW_CHAINS_FILES_PATH = `${RAW_FILES_PATH}/packages/config/src/chains`
-
-// for bridging package
-export const RAW_PROVIDERS_FILES_PATH = `${RAW_FILES_PATH}/src/bridging/providers` // TODO: fix to packages/bridging after copy
-export const DEFAULT_GAS_COST_FOR_HOOK_ESTIMATION = 110_000 // Based on https://dashboard.tenderly.co/anxolin/project/tx/0xc7d230c4119dabc7a4c332619a275d2390cf4596662be0524c04d1fa7a030c38/gas-usage
```
Duplicated elsewhere or no longer needed?

shoom3301 replied:
Duplicated in `packages/bridging/src/const.ts`

shoom3301 commented:
Thanks @jeffersonBastos !

1. See https://github.com/cowprotocol/cow-sdk/issues/388#issuecomment-3210064229
2. See https://github.com/cowprotocol/cow-sdk/issues/388#issuecomment-3210064229
3. Fixed
4. Fixed
5. Fixed
6. Fixed (and also changed in https://github.com/cowprotocol/cow-sdk/pull/426)
